core/src/main/java/hudson/util/io/ZipArchiver.java

```diff
@@ -48,12 +53,21 @@
     private final byte[] buf = new byte[8192];
     private final ZipOutputStream zip;
     private final OpenOption[] openOptions;
+    private final String prefix;
 
     ZipArchiver(OutputStream out) {
-        this(out, false);
+        this(out, false, "");
     }
 
-    ZipArchiver(OutputStream out, boolean failOnSymLink) {
+    // Restriction added for clarity, it's a package class, you should not use it outside of Jenkins core
+    @Restricted(NoExternalUse.class)
+    ZipArchiver(OutputStream out, boolean failOnSymLink, String prefix) {
+        if (StringUtils.isBlank(prefix)) {
+            this.prefix = "";
+        } else {
+            this.prefix = Util.ensureEndsWith(prefix, "/");
```

Comment:
Also trim it? Just to avoid surprises like "myprefix/ /" in user inputs

Reply:
Trim added. From my PoV there is no direct need "now" as the prefix is only coming from the baseFile at the moment but I agree this could be potentially useful for future usage, so better safe than sorry.

Thanks Oleg :)

Reply:
AFAIU, this conversation is 'resolved'? thanks